students/duanez2021/lesson03/Slice_Lab_01.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+import sys
+##############################################################
+# 20200617    djm list lab series 4
+#
+# Duane McCollum Python self-paced winter 2020
+#
+# slice lab
+# Write some functions that take a sequence as an argument, and return a copy of that sequence:
+#
+#     with the first and last items exchanged.
+#     with every other item removed.
+#     with the first 4 and the last 4 items removed, and then every other item in the remaining sequence.
+#     with the elements reversed (just with slicing).
+#     with the last third, then first third, then the middle third in the new order.
+#############################################################
+
+# List lab # 3
+fruit = list(('Apples', 'Pears', 'Oranges' ,'Peaches'))
+s = "I can't tell the difference between Whizzo butter and this dead crab."
+def swap_first_and_last(seq):
+    if type(seq) is not list:
+        #it's a string....
+        x = list(seq)
+        last_item = seq[len(seq) - 1]
+        first_item = seq[0]
+        x[0] = last_item
+        x[len(x)-1] = first_item
+        x = ''.join(x)
+    else:
+        #it's a not string....
+        x = seq[:]
+        last_item = seq[len(seq) - 1]
+        first_item = seq[0]
+        x[0] = last_item
+        x[len(x) - 1] = first_item
+    return (x)
```

Comment:
this problem should be solved with slicing only, there should be no use of index access or having to check the type

if you only use slicing, it will work for each data type (hint: it's a one liner)